client/allocrunner/alloc_runner.go

```diff
@@ -1205,19 +1227,55 @@ func (ar *allocRunner) GetTaskEventHandler(taskName string) drivermanager.EventH
 	return nil
 }
 
-// RestartTask signalls the task runner for the  provided task to restart.
-func (ar *allocRunner) RestartTask(taskName string, taskEvent *structs.TaskEvent) error {
+// Restart satisfies the WorkloadRestarter interface and restarts all tasks
+// that are currently running. Only the TaskRestartRunningSignal event type may
+// be used.
+func (ar *allocRunner) Restart(ctx context.Context, event *structs.TaskEvent, failure bool) error {
+	if event.Type != structs.TaskRestartRunningSignal {
+		return fmt.Errorf("Invalid event %s for alloc restart request", event.Type)
+	}
```

Comment:
I was wondering if it would be better to just overwrite `event.Type` here since I don't think it _really_ matters for callers, but I may be missing something so it would be good to hear from others.

Reply:
Normally we'd probably validate this at the RPC handler but in this case the same handler gets used for all three methods. That being said, we switch on the method based on `Event.Type`, right? So if it's wrong down here in the alloc runner that's a programmer's error where I'd expect it's fine to assert the right value via overwriting `event.Type`.